charms/worker/charmcraft.yaml

```diff
@@ -132,21 +132,27 @@ resources:
 
 parts:
   charm:
-    plugin: charm
-    build-packages: [git]
-    charm-entrypoint: k8s/src/charm.py
-    charm-requirements: [k8s/requirements.txt]
-  promote:
-    # move paths out of ./k8s to ./ since 
-    # charmcraft assumes ./lib to be there
-    # charmcraft assumes ./templates to be there
-    after: [charm]
-    plugin: nil
-    source: ./
-    override-prime: |
-      rm -rf $CRAFT_PRIME/lib $CRAFT_PRIME/templates
-      mv $CRAFT_PRIME/k8s/lib $CRAFT_PRIME/lib
-      mv $CRAFT_PRIME/k8s/templates $CRAFT_PRIME/templates
+    source: .
+    plugin: uv
+    build-packages:
+      - git
+      - libffi-dev
+      - libssl-dev
+      - pkg-config
+    build-snaps:
+      - astral-uv
+      - rustup
+    override-build: |
+      rustup default stable
+      for item in pyproject.toml uv.lock lib templates src; do
+        cp -r "$CRAFT_PART_BUILD/k8s/$item" "$CRAFT_PART_BUILD"
```

Comment:
why `copy` and not `move`?

Reply:
I've noticed some failures when we use `move`. This is likely because the build directory gets messed up, making it impossible to repack without cleaning the environment (i.e., we already have moved the files). However, this isn't a major problem as the prime step would only keep the `src` and `lib` folders (we add `templates` in the next step).